client/data/home/home-layout-flags.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import config from '@automattic/calypso-config';
+
+interface HomeLayoutFlags {
+	forcedView: string | null;
+	isDev: boolean;
+}
+
+export function getHomeLayoutFlags(): HomeLayoutFlags {
+	const pageParams = new URLSearchParams( document.location.search.substring( 1 ) );
```

Comment:
We have abstractions for routing so that we don't need to read query params directly from `document.location`.

There is the `context.query` object that the original code used to figure out the `isDev` and `forcedView` props.

And there's also the `getCurrentQueryArguments` Redux selector -- the current route information is also in Redux.

Reply:
I like your example below that uses the `getCurrentQueryArguments` selector 👍